src/cheri_insts.sail

```diff
@@ -229,6 +231,79 @@ function clause execute (CGetOffset(rd, cs1)) = {
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the **upper half** of capability
+ * register *cs1*.
+ *
+ * - The bits returned here are of the **in-register** form of the capability,
```

Comment:
I wonder if this should be the in-memory representation instead? That would make it easier to create capabilities with a known 128-bit representation. That would ensure that you get the same value back when you store some bits to the upper part of a capability, spill it to memory and load back using a integer loads.

Reply:
That's almost surely a better plan, treating these opcodes as export to / import from memory.  I'm not sure why I thought this was what we wanted, and https://github.com/CTSRD-CHERI/cheri-architecture/pull/78 is no help in jogging my memory.